.github/actions/generate-commit-list-v1/src/run.test.ts

```diff
@@ -0,0 +1,152 @@
+import sinon from 'sinon'
+import { assert } from 'chai'
+import type { Core } from './types'
+import * as exec from '@actions/exec'
+import run from './run'
+import {
+  expectedParsedCommitList,
+  rawCommitList,
+  rawRepositoryURL
+} from './test-utils'
+
+describe('run', () => {
+  let core: Core
+  let setFailed: sinon.SinonSpy
+  let setOutput: sinon.SinonSpy
+  let getInput: sinon.SinonStub
+  let info: sinon.SinonSpy
+  let getExecOutputStub: sinon.SinonStub
+
+  beforeEach(() => {
+    getExecOutputStub = sinon.stub(exec, 'getExecOutput')
+    setFailed = sinon.spy()
+    setOutput = sinon.spy()
+    getInput = sinon.stub()
+    info = sinon.spy()
+
+    core = {
+      setFailed,
+      setOutput,
+      getInput,
+      info
```

Comment:
One of the issues with spying/stubbing all properties is that you don't know if your code is calling things you didn't expect (e.g. you forgot a `return` to stop the code execution).